linux: Enable Control Group feature on systemd environment

Ponsuke07 opened the pull request:
# Purpose of pull request

Add support systemd of beaglebone black

# Test
## How to test

1. build beagleboneblack with systemd

```
MACHINE = "beaglebone"
DISTRO_FEATURES_append = " systemd"
DISTRO_FEATURES_BACKFILL_CONSIDERED += "sysvinit"
VIRTUAL-RUNTIME_init_manager = "systemd"
VIRTUAL-RUNTIME_initscripts = "systemd-compat-units"
```

2. Run beaglebone black

## Test result

If you do not have this fix, you will get the following error and failed to boot.

```
[    5.523808] systemd[1]: Failed to mount tmpfs at /sys/fs/cgroup: No such file or directory
[    5.540287] systemd[1]: Failed to mount cgroup at /sys/fs/cgroup/systemd: No such file or directory
[!!!!!!] Failed to mount API filesystems.
```

masami256 commented:
@Ponsuke07 Thank you for the patch !

I think cgroups isn't depend on devices, so it would be nice to move this setting from beaglebone.config to other file(e.g. resource-management.config). Then add following line(I haven't tested it) to linux-common.inc.

```
SRC_URI_append = " ${@oe.utils.conditional("VIRTUAL-RUNTIME_init_manager ", "systemd", " file://resource-management.config", "", d)}"
```

Or,  set new environment value to be able to use cgroups without systemd.

```
SRC_URI_append = " ${@oe.utils.conditional("SOME ENVIRONMENT VALUE NAME", "1", " file://resource-management.config", "", d)}"
```

Thereby user will have choice to use a cgroups or not.

hiraku-wfs commented:
"systemd-dependencies.config" would be a better name.

Ponsuke07 committed:
kernel: Add a kernel configuration file for systemd

Add a kernel configurations file for systemd.
This will be enabled only if systemd is selected as init_manager.

For more information about each configurations, please see below.
 * https://github.com/systemd/systemd/blob/e62a7fea757f259eb330da5b6d3ab4ede46400a2/README#L34

Ponsuke07 commented:
@masami256 @hiraku-wfs Thank you for reply and suggestion.     
     
I have created v2 applying your points. 
**Note that** I only tested this with beaglebone.      
I think we should make sure that this patch is not adversely affecting other boards.  
\# The kernel-configs add by "+=" has a lower priority than "_append". 
\# So I believe my PR will not cause any major problems, but just in case

masami256 commented on recipes-kernel/linux/files/systemd-dependencies.config:
```diff
@@ -0,0 +1,20 @@
+CONFIG_CRYPTO_USER_API_HASH=m
+CONFIG_CRYPTO_HMAC=m
+CONFIG_CRYPTO_SHA256=m
+# CONFIG_SYSFS_DEPRECATED is not set
```
Do we need commented lines?

Ponsuke07 replied:
This is necessary to override config(=y) in the based defconfig when running [merge_config()](https://github.com/meta-debian/meta-debian/blob/warrior/classes/merge-config.bbclass#L31).

masami256 replied:
thanks! I got it.

masami256 commented:
I changed PR title because it doesn't depend on specific boards.

masami256 commented:
@Ponsuke07 I tested this commit on qemuarm64 and it works fine. Thanks!

Ponsuke07 commented:
Thank you so much!
